### PR TITLE
Add custom bbb logo and css file

### LINKED ISSUE #68: Theming
**Is your feature request related to a problem? Please describe.**
With some bbb params, the room style can be modified. (https://docs.bigbluebutton.org/admin/customize.html#examplesn) We should take advantage of it and provide a css file that is handed over to the bbb room.
Ideally the css can be changed via the gui and effect the frontend and the bbb-room.

**Describe the solution you'd like**
Create two urls that provide the theming css for the frontend and bbb room.

**Describe alternatives you've considered**
On any theming change, create the two css files, write them to the hard drive and serve as static files (performance increase), that is the way greenlight handles it.

**Additional context**
*none*

### PR DESCRIPTION
**Pull request by SamuelWei**
# Fixes #68 

## Type (Highlight the corresponding type)
- Bugfix
- **Feature**
- Documentation
- Refactoring (e.g. Style updates, Test implementation, etc.)
- Other (please describe):

## Checklist
- [x] Code updated to current masters head
- [x] Passes CI checks
- [x] Is a part of an issue
- [x] Tests added for the bugfix or newly implemented feature, describe below why if not
- [x] Changelog is updated
- [x] Documentation of code and features exists

## Changes

- Add custom logo and css for BBB rom

## Other information
- Customization of the frontend (also part of #68) isn't addressed in this PR.

### CODE REVIEW

File: resources/js/views/settings/Application.vue [hunk 1]
```diff
@@ -819,6 +819,131 @@
...
+            <div v-html="fieldError('logo')"></div>
+            <div v-html="fieldError('logo_file')"></div>
+          </template>
+        </b-form-group>
+
+        <b-form-group
+          label-class="font-weight-bold"
+          class="mb-4"
+          :label="$t('settings.application.bbb.style.title')"
+          label-for='default_presentation'
```
Comment (dennis95stumm) [inline resources/js/views/settings/Application.vue:889]: This is wrong! This should be style or something like that!

File: resources/js/views/settings/Application.vue [hunk 2]
```diff
@@ -819,6 +819,131 @@
         </b-form-group>
 
         <hr>
+        <h4>{{ $t('settings.application.bbb.title') }}</h4>
+
+        <!--Logo Settings-->
+        <b-form-group
+          label-class="font-weight-bold"
+          class="mb-4"
+          label-for="application-logo-input"
```
Comment (dennis95stumm) [inline resources/js/views/settings/Application.vue:828]: This is wrong! This should be bbb-logo-input!

File: resources/js/views/settings/Application.vue [hunk 3]
```diff
@@ -819,6 +819,131 @@
...
+                v-b-tooltip
+                :title="$t('settings.application.bbb.style.reset')"
+              >
+                <i class="fas fa-undo"></i>
+              </b-button>
+              <!-- Delete file -->
+              <b-button
+                v-if='!viewOnly && bbb_style === null && !bbb_style_deleted'
+                variant='danger'
+                @click="bbb_style_deleted = true"
```
Comment (dennis95stumm) [inline resources/js/views/settings/Application.vue:930]: Why not just set `bbb_style` to `null`?

File: resources/js/views/settings/Application.vue [hunk 4]
```diff
@@ -853,16 +978,22 @@ export default {
...
       uploadLogoFileSrc: null,
+      uploadBBBLogoFile: null,
+      uploadBBBLogoFileSrc: null,
       uploadFaviconFile: null,
       uploadFaviconFileSrc: null,
       isBusy: false,
       default_presentation: null,
       default_presentation_deleted: false,
+      bbb_style: null,
+      bbb_style_deleted: false,
```
Comment (dennis95stumm) [inline resources/js/views/settings/Application.vue:989]: Why we need this variable?

**Review by dennis95stumm — CHANGES_REQUESTED** [review]

File: app/Http/Requests/UpdateSetting.php [hunk 5]
```diff
@@ -48,6 +48,9 @@ public function rules()
             'statistics.meetings.retention_period' => 'required|numeric|min:1|max:365',
             'attendance.enabled'                   => 'required|boolean',
             'attendance.retention_period'          => 'required|numeric|min:1|max:365',
+            'bbb.logo'                             => 'nullable|string|max:255',
+            'bbb.logo_file'                        => 'nullable|image|max:500',
+            'bbb.style'                            => 'nullable|file|max:500',
```
[the remark this answers is not in the document]
Comment (SamuelWei) [reply app/Http/Requests/UpdateSetting.php:53]: I tried to add a mime check, but css will always be detected as txt.

File: app/Http/Requests/UpdateSetting.php [hunk 6]
```diff
@@ -48,6 +48,9 @@ public function rules()
             'statistics.meetings.retention_period' => 'required|numeric|min:1|max:365',
             'attendance.enabled'                   => 'required|boolean',
             'attendance.retention_period'          => 'required|numeric|min:1|max:365',
+            'bbb.logo'                             => 'nullable|string|max:255',
+            'bbb.logo_file'                        => 'nullable|image|max:500',
```
[the remark this answers is not in the document]
Comment (SamuelWei) [reply app/Http/Requests/UpdateSetting.php:52]: The mime check if part of the 'image' validation rule

File: resources/js/views/settings/Application.vue (hunk 3 above)
Comment (SamuelWei) [reply resources/js/views/settings/Application.vue:930]: I think this is something you added for default presentation as well, to revert the deletion before submitting the form

Comment (dennis95stumm) [reply resources/js/views/settings/Application.vue:930]: Ah, yes.

**Pull request merged**